fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+!
+!-----------------------------------------------------------------------
+!***  SET THE FIRST WRITE GROUP AS THE FIRST ONE TO ACT.
+!-----------------------------------------------------------------------
+!
+      n_group = 1
+!
+!end quilting
+    endif
+!
+    ! --- advertise Fields in importState and exportState -------------------
+
+    if( cpl ) then
+
+        ! importable fields:
+        do i = 1, size(ImportFieldsList)
+          if (importFieldShare(i)) then
+            call NUOPC_Advertise(importState,         &
+                                 StandardName=trim(ImportFieldsList(i)), &
+                                 SharePolicyField="share", vm=fcstVM, rc=rc)
+            if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          else
+            call NUOPC_Advertise(importState,         &
+                                 StandardName=trim(ImportFieldsList(i)), vm=fcstVM, rc=rc)
+            if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          end if
+        end do
+      
+        ! exportable fields:
+        do i = 1, size(exportFieldsList)
+          if (exportFieldShare(i)) then
+            call NUOPC_Advertise(exportState,                            &
+                                 StandardName=trim(exportFieldsList(i)), &
+                                 SharePolicyField="share", vm=fcstVM, rc=rc)
+            if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          else
+            call NUOPC_Advertise(exportState,         &
+                                 StandardName=trim(exportFieldsList(i)), vm=fcstVM, rc=rc)
+            if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+          end if
+        end do
+      
+      if(mype==0) print *,'in fv3_cap, aft import, export fields in atmos'
+    endif
+
+    if(mype==0) print *,'in fv3_cap, init time=',mpi_wtime()-timeis
+!-----------------------------------------------------------------------
+!
+  end subroutine InitializeAdvertise
+
+  !-----------------------------------------------------------------------------
+
+  subroutine InitializeRealize(gcomp, importState, exportState, clock, rc)
+    type(ESMF_GridComp)  :: gcomp
+    type(ESMF_State)     :: importState, exportState
+    type(ESMF_Clock)     :: clock
+    integer, intent(out) :: rc
+
+    ! local variables
+    logical :: isPetLocal
+    integer :: n
+
+    rc = ESMF_SUCCESS
+
+    ! --- conditionally realize or remove Fields in importState and exportState -------------------
+
+    if ( cpl ) then
+
+      isPetLocal = ESMF_GridCompIsPetLocal(fcstComp, rc=rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__,  file=__FILE__)) return
+
+      if (isPetLocal) then
```

Comment:
@theurich -  should the below be inside a phase of the fcst grid comp?  And called from here - since only the forecast is realizing fields?